exposed-java-time/src/main/kotlin/org/jetbrains/exposed/sql/JavaDateFunctions.kt

```diff
@@ -25,12 +25,27 @@ class CurrentDateTime : Function<LocalDateTime>(JavaLocalDateTimeColumnType.INST
     }
 }
 
+class Year<T:Temporal?>(val expr: Expression<T>): Function<Int>(IntegerColumnType()) {
+    override fun toQueryBuilder(queryBuilder: QueryBuilder) = queryBuilder {
+        when (currentDialect) {
+            is PostgreSQLDialect -> append("EXTRACT(YEAR FROM ", expr, ")")
+            is OracleDialect -> append("EXTRACT(YEAR FROM ", expr, ")")
+            is MysqlDialect -> append("EXTRACT(YEAR FROM ", expr, ")")
+            is SQLServerDialect -> append("YEAR(", expr, ")")
+            is MariaDBDialect -> append("YEAR(", expr, ")")
+            is SQLiteDialect -> append("STRFTIME('%Y',", expr, ")")
+            is H2Dialect -> append("YEAR(", expr, ")")
+            else -> append("YEAR(", expr, ")")
```

Comment:
In the else statement i thought of two solutions:
- To throw an Exception telling that the current database dialect is not supprted. So if a new dialect is added we must add it's syntax here.
- Return Function-name(column-name) which can be a "default" syntax if a new dialect is added to Exposed.
So what you think is best to do?

Reply:
@hichem-fazai , sorry I was on a vacation and missed your comment. 
I guess what `UnsupportedByDialectException` should be fine.

Reply:
Hi @Tapac, ok perfect :)
I added a commit to handle the exception for all the functions.